src/modules/dmCommands/commands/settings.js

```diff
@@ -0,0 +1,103 @@
+        ]
+    );
+    pages.push(cooldownPage);
+
+    subscribe = userDto.notifications.get('twitter');
+    const twitterPage = new EmbedPage('twitter', 'Get notified when Jwiggs tweets', [
+        new MessageButton()
+            .setLabel(subscribe ? 'unsubscribe' : 'subscribe')
+            .setStyle(subscribe ? 'DANGER' : 'SUCCESS')
+            .setCustomId('settings.tweets'),
+    ]);
+    pages.push(twitterPage);
+
+    subscribe = userDto.notifications.get('levelPing');
+    const levelPingPage = new EmbedPage('Levelup Pings', 'Get notified when you level up', [
+        new MessageButton()
+            .setLabel(subscribe ? 'unsubscribe' : 'subscribe')
+            .setStyle(subscribe ? 'DANGER' : 'SUCCESS')
+            .setCustomId('settings.levelPing'),
+    ]);
+    pages.push(levelPingPage);
+
+    const member = await getMember(user.id);
+    const osPage = new EmbedPage(
+        'Operating System',
+        'People use different Operating System. Choose the one you use below',
+        [
+            new MessageButton({
+                customId: `settings.os.windows`,
+                style: member.roles.cache.has(getRole('windows')) ? 'DANGER' : 'SUCCESS',
+                label: 'Windows (as useful as in space)',
+            }),
+            new MessageButton({
+                customId: `settings.os.linux`,
+                style: member.roles.cache.has(getRole('linux')) ? 'DANGER' : 'SUCCESS',
+                label: 'Linux (penguin gang)',
+            }),
+            new MessageButton({
+                customId: `settings.os.mac`,
+                style: member.roles.cache.has(getRole('mac')) ? 'DANGER' : 'SUCCESS',
+                label: 'Mac (with cheese please)',
+            }),
+        ]
+    );
+    pages.push(osPage);
+
+    subscribe = member.roles.cache.has(getRole('subscriber'));
+    const subscriberPage = new EmbedPage('Subscriber', 'Get Pinged, when Announcements are made', [
+        new MessageButton()
+            .setLabel(subscribe ? 'unsubscribe' : 'subscribe')
+            .setStyle(subscribe ? 'DANGER' : 'SUCCESS')
+            .setCustomId('settings.subscriber'),
+    ]);
+    pages.push(subscriberPage);
+
+    subscribe = member.roles.cache.has(getRole('tester'));
+    const testerPage = new EmbedPage('Tester', 'Get access to test versions of the game', [
+        new MessageButton()
+            .setLabel(subscribe ? 'unsubscribe' : 'subscribe')
+            .setStyle(subscribe ? 'DANGER' : 'SUCCESS')
+            .setCustomId('settings.tester'),
+    ]);
+    pages.push(testerPage);
+
+    return pages;
+}
+
+/**
+ *
+ * @param {import('discord.js').Client} client
+ * @param {import('discord.js').Message} message
+ * @return {Promise<void>}
+ */
+export async function fun(client, message) {
+    const embedPages = await getPages(message.author);
+    const embedPageMessage = new EmbedPageMessage(embedPages);
+    await embedPageMessage.update(message);
```

Comment:
Maybe just take in the message at constructor so no need to call update.

Reply:
update is async. afaik a constructor must not be async

Reply:
It doesn't need to be async if you don't await the sending

Reply:
Alternitivly pass the messsage to the constructor than make a static prop to create a class where you pass args other than message and then it sends message passes args and message and gives you class

Reply:
by awaiting one ensures that we have a trace of the error if something happens within the await.